src/s3ql/backends/gs.py

```diff
@@ -101,7 +101,11 @@ def __init__(self, ssl_context: ssl.SSLContext):
         self.conn_pool = dict()  # type: Dict[Tuple[str,int], HTTPConnection]
 
     def  __call__(self, url:str, method: str = 'GET',
-                  body : Optional[bytes] = None, headers=None, timeout=None):
+                  body : Union[bytes, str, None] = None, headers=None, timeout=None):
+
+        if type(body) == str:
+            body = cast(str, body)
+            body = body.encode('utf-8')
```

Comment:
Ugh, is this truly needed? S3QL is in no-position to decide what encoding the remote server expects.

Reply:
Yeah, I know ... I am not sure to why I was being passed a string from the google api making the request.

Reply:
@Nikratio The offending line is here https://github.com/googleapis/google-auth-library-python/blob/137b43ba002d422c68e7bfacdafa54986f38500b/google/oauth2/_client.py#L100

So If I understand correctly, all characters are ascii encoded.

The body object passed in comes from here:

https://github.com/googleapis/google-auth-library-python/blob/137b43ba002d422c68e7bfacdafa54986f38500b/google/oauth2/_client.py#L230